kobo/apps/trash_bin/models/project.py

```diff
@@ -83,4 +86,11 @@ def toggle_statuses(
                 )
                 assert updated >= kc_updated
 
+                # Update attachment storage counters
+                attachments = Attachment.objects.filter(
+                    xform__kpi_asset_uid__in=object_identifiers
+                )
+                bulk_update_attachment_storage_counters(
+                    attachments, subtract=not active
+                )
```

Comment:
Beware, I think the calculation always takes place whatever the value of `toggle_delete`. 
When `toggle_delete` is False, projects are only archived. The storage should not be affected. If the boolean `toggle_delete` is not obvious enough, do not hesitate to rename it and add explanation in the docstring how it should be used. i.e. 

`active`| `toggle_delete` | behaviour
--- | --- | --- 
False | False | Archive project
False | True | Prepare project for soft-deletion (`move_to_trash` needs to be called right after)
True | False | Unarchive project
True | True | Prepare project to be restored from trash (`put_back` needs to be called right after)